### PR TITLE
Fix clear value for relational interfaces

### LINKED ISSUE #12854: Clear value causing error for relational fields
### Describe the Bug

(This is more of a follow up to #12847, but moving this to a new issue as it's more about null values in general rather than singletons now.)

When we click "clear value" in relational interfaces, it will cause error:

https://user-images.githubusercontent.com/42867097/164031716-fcccf683-b3a5-475e-bdf8-0868fa801241.mp4

Tested M2M, M2A, O2M and Files interface, but should also affect Images etc.

A potential solution was changing this line to `props.value ?? []`:

https://github.com/directus/directus/blob/5cce44fa0547a7e4e6b04e479da245457b238647/app/src/interfaces/list-o2m/list-o2m.vue#L130

but it would still make the field be "edited", as the default raw value is expected to be `undefined` with the prop falling back to `[]`.

### To Reproduce

1. Create a "to many" relational field, such as M2M, M2A, O2M or Files.
2. Add an item to the field so that we can click "Clear value" later.
3. Click on "clear value" which sets the field to `null`, and causes the error.

### Errors Shown

Refer clip above.

### What version of Directus are you using?

9.9.0 (latest branch)

### What version of Node.js are you using?

16.13.0

### What database are you using?

PostgreSQL 13.5

### What browser are you using?

Chrome

### How are you deploying Directus?

running locally

### PR DESCRIPTION
**Pull request by azrikahar**
Fixes #12854

## Problem

Currently "to many" relational interfaces (or ones using `useRelationMultiple`) requires the initial payload to at least be an empty array. When the `props.value` is undefined, it is defaulted to an empty array as well.

However when we use Clear Value, it emits `null` value (not undefined), which breaks them:

https://user-images.githubusercontent.com/42867097/167824367-6f134567-4b13-451d-985d-aee486eee713.mp4

## Solution

Now useRelationMultiple will clear the value properly, and ensure it being `undefined` or `[]` depending on the initial value to not stage any edits. 

However if there are any existing values (not newly created or updated ones, those will still be cleared), we will stage them for deletion instead.

https://user-images.githubusercontent.com/42867097/167826057-7a18fdee-3a1f-4836-a251-dbc85b208aa7.mp4

### CODE REVIEW

File: app/src/interfaces/list-o2m/list-o2m.vue [hunk 1]
```diff
@@ -87,7 +87,7 @@
 <script setup lang="ts">
 import { useRelationO2M, useRelationMultiple, RelationQueryMultiple, DisplayItem } from '@/composables/use-relation';
 import { parseFilter } from '@/utils/parse-filter';
-import { Field, Filter } from '@directus/shared/types';
+import { Filter } from '@directus/shared/types';
```
Comment (azrikahar) [inline app/src/interfaces/list-o2m/list-o2m.vue:90]: Only removing an unused variable here 👍

**Comment by rijkvanzanten** [comment]
> However if there are any existing values (not newly created or updated ones, those will still be cleared), we will stage them for deletion instead.

I'm wondering if this is needed. The API will also clear everything if `null` is submitted IIRC, so we might be able to just have the app handle `null` as `[]`, and be done 🤔

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]
> However if there are any existing values (not newly created or updated ones, those will still be cleared), we will stage them for deletion instead.

I'm wondering if this is needed. The API will also clear everything if null is submitted IIRC, so we might be able to just have the app handle null as [], and be done 🤔

**Comment by azrikahar** [comment]
> > However if there are any existing values (not newly created or updated ones, those will still be cleared), we will stage them for deletion instead.
> 
> I'm wondering if this is needed. The API will also clear everything if null is submitted IIRC, so we might be able to just have the app handle null as [], and be done 🤔

It was initially due to the existing items persisting visually after being cleared:

https://user-images.githubusercontent.com/42867097/168288421-f8a4563c-5f51-4522-a2a2-6e0a48d57492.mp4

hence I opted to use the current approach. I do agree it'll feel more like the previous interaction if we fully clear the items, so I tried to implement it here but there's a slight issue where the Add Existing drawer will still filter out existing items:

https://user-images.githubusercontent.com/42867097/168289439-ef8f3276-bbf8-457a-969b-48b1ec2b667f.mp4

It's due to the following filter to exclude an existing entry's relational items:

https://github.com/directus/directus/blob/c30b041654af1111b8bb3cbdf2200c790032958d/app/src/interfaces/list-m2m/list-m2m.vue#L291-L299

So not sure how to "toggle" this off after a user clears the value to full empty to ensure they can select the existing items again, at least without even more changes I presume 🤔 (and will need to "toggle" this on back again when user uses Undo Value)

File: app/src/composables/use-relation/use-relation-multiple.ts [hunk 2]
```diff
@@ -270,6 +274,26 @@ export function useRelationMultiple(
...
+			}
+
+			const pkField =
+				relation.value.type === 'o2m'
+					? relation.value.relatedPrimaryKeyField.field
+					: relation.value.junctionPrimaryKeyField.field;
+
+			target.value.create = [];
+			target.value.update = [];
+			target.value.delete = fetchedItems.value.map((item) => item[pkField]);
```
Comment (rijkvanzanten) [inline app/src/composables/use-relation/use-relation-multiple.ts:292]: Should clear stage every item for deletion, or should this be `[]` as well with a call elsewhere to refresh the fetched items?

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**